projects/vdk-plugins/vdk-jupyter/vdk-jupyterlab-extension/vdk_jupyterlab_extension/handlers.py

```diff
@@ -11,6 +12,22 @@
 from .vdk_ui import VdkUI
 
 
+class HandlerConfiguration:
+    def __init__(self):
+        self._rest_api_url = os.environ["REST_API_URL"]
+        if not self._rest_api_url:
+            raise Exception(
+                "Missing environment variable REST_API_URL. "
+                "This is probably caused by a corrupt environment."
```

Comment:
Here you can add more information on what the rest api url  is for (or where it is used) and consequences of the missing variable, to make it more user accessible.

Reply:
maybe I'm wrong about this, but I don't think a user should ever be thinking about rest api URLs, it should be seamless; this error isn't something we expect users to be hitting ever

Reply:
I think you are absolutely right, that in best case, users should not need to think about rest API URLs and should be seamless. 

But if they hit the error then what ?

In this case we should use clear, concise, and informative error messages to help users understand what went wrong and how to resolve it. (As outlined in https://github.com/vmware/versatile-data-kit/wiki/Design-Principles) 

For that reason our error handling format recommends that a good error message should answer what happened, why it happened , what are the consequences for the user and what are the countermeasures (how the user can resolve it). 
([[link](https://github.com/vmware/versatile-data-kit/blob/main/projects/control-service/CONTRIBUTING.md#error-handling)]) 
Let's try each error message and exception message to answer those questions.